homeassistant/components/statistics/sensor.py

```diff
@@ -264,6 +277,26 @@ def _purge_old(self):
                 self.min_age = self.max_age = dt_util.utcnow()
                 self.change = self.average_change = STATE_UNKNOWN
                 self.change_rate = STATE_UNKNOWN
+        self.async_schedule_update_ha_state()
+
+        # If max_age is set, ensure to update again after the defined interval.
+        next_to_purge_timestamp = self._next_to_purge_timestamp()
+        if next_to_purge_timestamp:
+            _LOGGER.debug(
+                "%s: scheduling update at %s.", self.entity_id, next_to_purge_timestamp
+            )
+            if self._update_listener:
+                self._update_listener()
+                self._update_listener = None
+
+            async def _scheduled_update(now):
+                """Timer callback for sensor update."""
+                _LOGGER.debug("%s: executing scheduled update", self.entity_id)
+                await self.async_update()
```

Comment:
Call `self.async_schedule_update_ha_state(True)`. That will call `async_update` and then write it.

Reply:
OK